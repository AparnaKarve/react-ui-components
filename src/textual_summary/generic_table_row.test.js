import React from 'react';
import renderer from 'react-test-renderer';
import GenericTableRow from './generic_table_row';

describe('Simple Table', () => {
  it('renders just fine...', () => {
    /*
     *  row has
     *    1) label
     *    2) icon
     *    3) value
     *  item.value
     *    a) simple
     *    b) array    TODO
     *  linkClick handler TODO
     */
    const item = {
      label: 'Network Manager',
      value: 'Amazon East Network Manager',
      icon: null,
      image: 'http://localhost:3000/assets/svg/vendor-ec2_network-04c432db85be0fd670cd6da83b8685dab51e1b7a63258b70cccbdf8d7f72c988.svg',
      link: '/ems_network/10000000000045',
      title: 'Show Network Manager \'Amazon East Network Manager\'',
      hoverClass: '',
    };

    const component = renderer.create(<GenericTableRow item={item} />);
    const row = component.toJSON();
    expect(row).toMatchSnapshot();
  });
});

