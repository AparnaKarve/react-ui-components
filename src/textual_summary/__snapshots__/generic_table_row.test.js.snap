// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Table renders just fine... 1`] = `
<tr
  className=""
  onClick={[Function]}
  title="Show Network Manager 'Amazon East Network Manager'"
>
  <td
    className="label"
  >
    Network Manager
  </td>
  <td>
    <img
      alt="Show Network Manager 'Amazon East Network Manager'"
      src="http://localhost:3000/assets/svg/vendor-ec2_network-04c432db85be0fd670cd6da83b8685dab51e1b7a63258b70cccbdf8d7f72c988.svg"
      title="Show Network Manager 'Amazon East Network Manager'"
    />
    <span>
       
      Amazon East Network Manager
    </span>
  </td>
</tr>
`;
